agent/agent.go

```diff
@@ -177,6 +184,10 @@ func (a *Agent) cacheExecutableArtifact(req *agentapi.DeployRequest) (*string, e
 	fileName := fmt.Sprintf("workload-%s", *a.md.VmID)
 	tempFile := path.Join(os.TempDir(), fileName)
 
+	if strings.EqualFold(runtime.GOOS, "windows") && strings.EqualFold(*req.WorkloadType, "elf") {
```

Comment:
I think it would be safer and more predictable if we avoided these kinds of runtime queries and instead used build tags and, better yet, cross-platform libraries, to avoid all of these branches.

Reply:
Not a fan of this pattern at all. Will make all of these instances live behind build tags instead, although this may be a little uglier.

Reply:
To be addressed in a subsequent, small PR ;)